libs/language-server/src/grammar/blocktype.langium

```diff
@@ -5,18 +5,38 @@
 import './expression'
 import './iotype'
 import './terminal'
+import './transform'
 import './valuetype'
+import './main'
+
+ReferenceableBlocktypeDefinition:
+  BuiltinBlocktypeDefinition | CompositeBlocktypeDefinition;
 
 BuiltinBlocktypeDefinition:
     'builtin' 'blocktype' name=ID '{'
         (inputs+=BlocktypeInput | outputs+=BlocktypeOutput | properties+=BlocktypeProperty)*
 '}';
 
+CompositeBlocktypeDefinition:
+    'composite' 'blocktype' name=ID '{'
+        (
+            inputs+=BlocktypeInput
+            | outputs+=BlocktypeOutput
+            | properties+=BlocktypeProperty
+            | blocks+=BlockDefinition
+            | pipes+=BlocktypePipeline
+            | transforms+=TransformDefinition
+        )*
+'}';
+
 BlocktypeInput:
     'input' name=ID 'oftype' iotype=[IotypeDefinition] ';';
 
 BlocktypeOutput:
     'output' name=ID 'oftype' iotype=[IotypeDefinition] ';';
 
 BlocktypeProperty:
-    'property' name=ID 'oftype' valuetype=ValuetypeReference (':' defaultValue=Expression)? ';';
\ No newline at end of file
+    'property' name=ID 'oftype' valuetype=ValuetypeReference (':' defaultValue=Expression)? ';';
+
+BlocktypePipeline:
+    input=[BlocktypeInput] '->' (blocks+=[BlockDefinition] '->')+ output=[BlocktypeOutput] ';';
```

Comment:
Just as a remark 'cause I'm unsure if you are aware: This grammar prevents splitting the pipeline into multiple statements like we allow for pipelines:

```
input1 -> IntermediateBlock;
IntermediateBlock -> output1;
```

I believe this "feature" is not required for the first version, but we should have it on the radar.